src/modules/pairs.ts

```diff
@@ -4,140 +4,167 @@ import { ZERO_BD, ZERO_BI } from "../utils/constants";
 import { getDayTotalTimestamp, getHourTotalTimestamp } from "../utils/timeframeTimestamp";
 
 export namespace pairs {
-    export function createOrUpdatePair(timestamp: BigInt, buyTokenId: string, sellTokenId: string, buyAmount: BigInt, sellAmount: BigInt, sellAmountEth: BigDecimal, sellAmountUsd: BigDecimal): void {
-        let canonicalMarket = getCanonicalMarket(buyTokenId, sellTokenId, buyAmount, sellAmount)
-        
-        let token0 = canonicalMarket.get("token0")
-        let volumeToken0 = BigInt.fromString(canonicalMarket.get("volumeToken0"))
-        let token1 = canonicalMarket.get("token1")
-        let volumeToken1 = BigInt.fromString(canonicalMarket.get("volumeToken1"))
-
-        let pairTotal = getOrCreatePair(token0, token1)
-        let pairDailyTotal = getOrCreatePairDaily(token0, token1, timestamp)
-        let pairHourlyTotal = getOrCreatePairHourly(token0, token1, timestamp)
-
-        totalsUpdate(pairTotal, pairDailyTotal, pairHourlyTotal, volumeToken0, volumeToken1, sellAmountEth, sellAmountUsd)
+
+  export class TokenProps {
+    token: string
+    volume: BigInt
+    price: BigDecimal
+		constructor(
+      _token: string,
+      _volume: BigInt,
+      _price: BigDecimal,
+    ) {
+      this.token = _token
+      this.volume = _volume
+      this.price = _price
+    }
+  }
+
+  export function createOrUpdatePair(timestamp: BigInt, buyTokenId: string, sellTokenId: string, buyAmount: BigInt, sellAmount: BigInt,
+    sellAmountEth: BigDecimal, sellAmountUsd: BigDecimal, buyTokenPrice: BigDecimal, sellTokenPrice: BigDecimal): void {
+    let canonicalMarket = getCanonicalMarket(buyTokenId, sellTokenId, buyAmount, sellAmount, buyTokenPrice, sellTokenPrice)
```

Comment:
@GabrielCamba im not sure if i understand here, but if i follow the logic correctly. This `sellTokenPrice` is in reality the USD price.

https://github.com/cowprotocol/subgraph/blob/2467789a7347c8cd8ee52d23325fff6c3cc51c37/src/modules/trades.ts#L46

![image](https://user-images.githubusercontent.com/2352112/168860529-29030375-b8f2-40a5-a4eb-a84043a49548.png)


1st concern, is our naming convention. I feel that if the price is USD we should say it in the name. 

2nd concern is that if I understand this code, you are using this price later https://github.com/cowprotocol/subgraph/pull/24/files#diff-26dee9f91cd639f585c0d6e3f6a0d634300c0a86c6ec48b913c49d699ff5b43fR30

https://github.com/cowprotocol/subgraph/blob/2467789a7347c8cd8ee52d23325fff6c3cc51c37/src/modules/pairs.ts#L41

Is it possible you use the "Usd" price as if it was the price of the token expressed in the other token of the Pair?

Reply:
> Is it possible you use the "Usd" price as if it was the price of the token expressed in the other token of the Pair?

I will work on this.

Reply:
Is this topic still open?

Reply:
I think this is already fixed.